server/pkcs11.go

```diff
@@ -0,0 +1,165 @@
+package server
+
+import (
+	"fmt"
+	"crypto"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/cbroglie/crypto11"
+	"github.com/cloudflare/cfssl/log"
+)
+
+// PKCS11URI defines a PKCS#11 URI: https://tools.ietf.org/html/rfc7512#section-2.3
```

Comment:
That's why I put a link to the RFC up there, because all of these are defined and used in there and if there is any confusion about what parameter X does you should refer to that RFC.